plugins/vertexai/modelgauge/suts/vertexai_mistral_sut.py

```diff
@@ -0,0 +1,111 @@
+    # the following fields required, so the user/client is forced to
+    # affirmatively specify them (for reproducibility)
+    temperature: Optional[float] = None
+    response_format: Optional[Dict[str, str]] = None
+    safe_prompt: Optional[bool] = True
+    stream: Optional[bool] = False
+    n: int = 1
+    max_tokens: Optional[int]
+
+
+class VertexAIMistralResponse(BaseModel):
+    model_config = ConfigDict(extra="ignore")
+
+    id: str
+    object: str
+    model: str
+    created: int
+    choices: list[Dict]
+    usage: Dict[str, int]
+
+
+@modelgauge_sut(capabilities=[AcceptsTextPrompt])
+class VertexAIMistralAISut(PromptResponseSUT):
+    """A MistralAI SUT hosted on GCP's Vertex service."""
+
+    def __init__(
+        self,
+        uid: str,
+        model_name: str,
+        model_version: str,
+        api_key: VertexAIAPIKey,
+        project_id: VertexAIProjectId,
+        region: VertexAIRegion,
+    ):
+        super().__init__(uid)
+        self.model_name = model_name
+        self.model_version = model_version
+        self._api_key = api_key
+        self._project_id = project_id
+        self._region = region
+        self._client = None
+
+    @property
+    def client(self) -> VertexAIClient:
+        if not self._client:
+            self._client = VertexAIClient(
+                publisher="mistralai",
+                model_name=self.model_name,
+                model_version=self.model_version,
+                streaming=False,
+                api_key=self._api_key,
+                project_id=self._project_id,
+                region=self._region,
+            )
+        return self._client
+
+    def translate_text_prompt(self, prompt: TextPrompt) -> VertexAIMistralRequest:
+        args = {"model": self.model_name, "messages": [{"role": _USER_ROLE, "content": prompt.text}]}
+        if prompt.options.temperature is not None:
+            args["temperature"] = prompt.options.temperature
+        if prompt.options.max_tokens is not None:
+            args["max_tokens"] = prompt.options.max_tokens
+        return VertexAIMistralRequest(**args)
+
+    def evaluate(self, request: VertexAIMistralRequest) -> VertexAIMistralResponse:
+        response = self.client.request(request.model_dump(exclude_none=True))  # type: ignore
+        return VertexAIMistralResponse(**response)
+
+    def translate_response(self, request: VertexAIMistralRequest, response: VertexAIMistralResponse) -> SUTResponse:
+        completions = []
+        for choice in response.choices:
+            text = choice["message"]["content"]
+            assert text is not None
+            completions.append(SUTCompletion(text=text))
+        return SUTResponse(completions=completions)
+
+
+VERTEX_KEY = InjectSecret(VertexAIAPIKey)
+VERTEX_PROJECT_ID = InjectSecret(VertexAIProjectId)
+VERTEX_REGION = InjectSecret(VertexAIRegion)
+
+model_name = "mistral-large"
+model_version = "2407"
+model_uid = f"{model_name}-{model_version}"
```

Comment:
I think @bollacker has mentioned in the past that the provider should also be a part of the UID. Especially if we supply multiple providers for one SUT.